tfjs-backend-webgpu/src/draw_test.ts

```diff
@@ -0,0 +1,194 @@
+    // Premultiplied data.
+    const data =
+        [.1, .2, .3, .4, .5, .6, .7, .8, .09, .1, .11, .12, .13, .14, .15, .16];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 4]);
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    const actualData =
+        await readPixelsFromGPUCanvas(canvas, height, width, unmultiplyAlpha);
+    expectArraysClose(actualData, data, 0.01);
+  });
+
+  it('draw image with 3 channels and int values', async () => {
+    const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 3], 'int32');
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    expectArraysEqual(
+        await readPixelsFromGPUCanvas(canvas, height, width, removeLastChannel),
+        data);
+  });
+
+  it('draw image with 3 channels and float values', async () => {
+    const data = [.1, .2, .3, .4, .5, .6, .7, .8, .9, .1, .11, .12];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 3]);
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    const actualData =
+        await readPixelsFromGPUCanvas(canvas, height, width, removeLastChannel);
+    const expectedData = [25, 51, 76, 102, 128, 153, 178, 204, 229, 25, 28, 31];
+    // On macOs/M2, some difference is close to 1.
+    expectArraysClose(actualData, expectedData, 1);
+  });
+
+  it('draw 2D image in grayscale', async () => {
+    const data = [1, 2, 3, 4];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor2d(data, [width, height], 'int32');
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    const actualData = await readPixelsFromGPUCanvas(canvas, height, width);
+    const expectedData =
+        [1, 1, 1, 255, 2, 2, 2, 255, 3, 3, 3, 255, 4, 4, 4, 255];
+    expectArraysEqual(actualData, expectedData);
+  });
+
+  it('draw image with alpha=0.5', async () => {
+    const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    const width = 6;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 1], 'int32');
+    const canvas = getCanvas();
+
+    const drawOptions = {
+      contextOptions: {contextType: env.name},
+      imageOptions: {alpha: 0.5}
+    };
+    // tslint:disable-next-line:no-any
+    tf.browser.draw(img, canvas as any, drawOptions);
+    const actualData = await readPixelsFromGPUCanvas(canvas, height, width);
+    const expectedData = [
```

Comment:
I am confused that why the `expectedData` is not `[1, 1, 1, 128, 2, 2, 2, 128, 3, 3, 3, 128, 4, 4, 4, 128 ...`. I asked @hujiajie. He told me that the webgpu should manually do it in shader. In that case, we can get the expected data and don't need an extra `unmultiplyAlpha` process. Please sync with @hujiajie to make sure the shader is written as expected. Thanks.